examples/connext_dds/asynchronous_publication_statistic/c/async_subscriber.c

```diff
@@ -0,0 +1,304 @@
+    }
+
+    /* To customize subscriber QoS, use
+       the configuration file USER_QOS_PROFILES.xml */
+    subscriber = DDS_DomainParticipant_create_subscriber(
+            participant,
+            &DDS_SUBSCRIBER_QOS_DEFAULT,
+            NULL /* listener */,
+            DDS_STATUS_MASK_NONE);
+    if (subscriber == NULL) {
+        printf("create_subscriber error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* Register the type before creating the topic */
+    type_name = asyncTypeSupport_get_type_name();
+    retcode = asyncTypeSupport_register_type(participant, type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        printf("register_type error %d\n", retcode);
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize topic QoS, use
+       the configuration file USER_QOS_PROFILES.xml */
+    topic = DDS_DomainParticipant_create_topic(
+            participant,
+            "Example async",
+            type_name,
+            &DDS_TOPIC_QOS_DEFAULT,
+            NULL /* listener */,
+            DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        printf("create_topic error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* Set up a data reader listener */
+    reader_listener.on_requested_deadline_missed =
+            asyncListener_on_requested_deadline_missed;
+    reader_listener.on_requested_incompatible_qos =
+            asyncListener_on_requested_incompatible_qos;
+    reader_listener.on_sample_rejected = asyncListener_on_sample_rejected;
+    reader_listener.on_liveliness_changed = asyncListener_on_liveliness_changed;
+    reader_listener.on_sample_lost = asyncListener_on_sample_lost;
+    reader_listener.on_subscription_matched =
+            asyncListener_on_subscription_matched;
+    reader_listener.on_data_available = asyncListener_on_data_available;
+
+    /* To customize data reader QoS, use
+       the configuration file USER_QOS_PROFILES.xml */
+    reader = DDS_Subscriber_create_datareader(
+            subscriber,
+            DDS_Topic_as_topicdescription(topic),
+            &DDS_DATAREADER_QOS_DEFAULT,
+            &reader_listener,
+            DDS_STATUS_MASK_ALL);
+    if (reader == NULL) {
+        printf("create_datareader error\n");
+        subscriber_shutdown(participant);
+        return -1;
+    }
+
+    /* Main loop */
+    for (count = 0; (sample_count == 0) || (count < sample_count); ++count) {
+        printf("async subscriber sleeping for %d sec...\n", poll_period.sec);
+        NDDS_Utility_sleep(&poll_period);
+
+        retcode = DDS_DataReader_get_datareader_protocol_status(
+                reader,
+                &protocol_status);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "get_datareader_protocol_status error\n");
+            subscriber_shutdown(participant);
+            return -1;
+        }
+        printf("Asynchronous Publisher Statistics:\n"
```

Comment:
Same comment as in the publisher code, if you think this it is clearer like this, to only include a subset of the stats that is fine. Otherwise add the others

Reply:
I have added:
```
        printf("Fragmented Data Statistics:\n"
                "\t received_fragment_count %lld\n"
                "\t dropped_fragment_count %lld\n"
                "\t sent_nack_fragment_count %lld\n"
                "\t sent_nack_fragment_bytes %lld\n",
                protocol_status.received_fragment_count,
                protocol_status.dropped_fragment_count,
                protocol_status.sent_nack_fragment_count,
                protocol_status.sent_nack_fragment_bytes);
```